scripts/write-config.sh

```diff
@@ -0,0 +1,206 @@
+#   --------------------------+-------------------------------------------------------------------------
+#   db-password.txt           | Database password
+#   --------------------------+-------------------------------------------------------------------------
+#   db-username.txt           | Database username
+#   --------------------------+-------------------------------------------------------------------------
+#   janitor-client-sa.json    | Creds to access Janitor service
+#   --------------------------+-------------------------------------------------------------------------
+#   sqlproxy-sa.json          | SA of the CloudSQL proxy
+#   --------------------------+-------------------------------------------------------------------------
+#   stairway-db-name.txt      | Stairway database name
+#   --------------------------+-------------------------------------------------------------------------
+#   stairway-db-password.txt  | Stairway database password
+#   --------------------------+-------------------------------------------------------------------------
+#   stairway-db-username.txt  | Stairway database username
+#   --------------------------+-------------------------------------------------------------------------
+#   target.txt                | the target that generated this set of config files. Allows the script 
+#                             | to skip regenerating the environment on a rerun.
+#   --------------------------+-------------------------------------------------------------------------
+#   testrunner-sa.json        | SA for running TestRunner
+#   --------------------------+-------------------------------------------------------------------------
+#   user-delegated-sa.json    | Firecloud SA used to masquerade as test users
+#   --------------------------+-------------------------------------------------------------------------
+#   wsm-sa.json               | SA that the WSM application runs as
+#   --------------------------+-------------------------------------------------------------------------
+
+function usage {
+  cat <<EOF
+Usage: $0 [<target>] [<vaulttoken>] [<outputdir>]"
+
+  <target> can be:
+    local - for testing against a local (bootRun) WSM
+    dev - uses secrets from the dev environment
+    alpha - alpha test environment
+    staging - release staging environment
+    help or ? - print this help
+    clean - removes all files from the output directory
+    * - anything else is assumed to be a personal environment using the terra-kernel-k8s
+  If <target> is not specified, then use the envvar WSM_WRITE_CONFIG_TARGET
+  If WSM_WRITE_CONFIG_TARGET is not specified, then use local
+
+  <vaulttoken> defaults to the token found in ~/.vault-token.
+
+  <outputdir> defaults to "config/", so when run in the gradle rootdir, it will be
+  in the expected place for automation.
+EOF
+ exit 1
+}
+
+# Get the inputs with defaulting
+default_target=${WSM_WRITE_CONFIG:-local}
```

Comment:
Nit: I think there's a mismatch between the envvar name here (`WSM_WRITE_CONFIG`) and the envvar name alluded to in the documentation (`WSM_WRITE_CONFIG_TARGET`)

Reply:
Good catch!